bayesian/v2/api_v2.py

```diff
@@ -0,0 +1,248 @@
+                _session.post(url=METRICS_SERVICE_URL + "/api/v1/prometheus", json=metrics_payload)
+                raise HTTPError(400, msg)
+        package = case_sensitivity_transform(ecosystem, package)
+
+        # Perform Component Analyses on Vendor specific Graph Edge.
+        analyses_result = VendorAnalyses(ecosystem, package, version).get_vendor_analyses()
+
+        if analyses_result is not None:
+            # Unknown component for Fabric8 Analytics
+            server_create_component_bookkeeping(ecosystem, package, version, g.decoded_token)
+
+            metrics_payload.update({"status_code": 200, "value": time.time() - st})
+            _session.post(url=METRICS_SERVICE_URL + "/api/v1/prometheus", json=metrics_payload)
+            return analyses_result
+
+        if os.environ.get("INVOKE_API_WORKERS", "") == "1":
+            # Trigger the unknown component ingestion.
+            server_create_analysis(ecosystem, package, version, user_profile=g.decoded_token,
+                                   api_flow=True, force=False, force_graph_sync=True)
+            msg = f"Package {ecosystem}/{package}/{version} is unavailable. " \
+                  "The package will be available shortly," \
+                  " please retry after some time."
+
+            metrics_payload.update({"status_code": 202, "value": time.time() - st})
+            _session.post(url=METRICS_SERVICE_URL + "/api/v1/prometheus", json=metrics_payload)
+
+            return response_template({'error': msg}, 202)
+
+        # No data has been found and INVOKE_API_WORKERS flag is down.
+        server_create_analysis(ecosystem, package, version, user_profile=g.decoded_token,
+                               api_flow=False, force=False, force_graph_sync=True)
+        msg = f"No data found for {ecosystem} package {package}/{version}"
+
+        metrics_payload.update({"status_code": 404, "value": time.time() - st})
+        _session.post(url=METRICS_SERVICE_URL + "/api/v1/prometheus", json=metrics_payload)
+
+        raise HTTPError(404, msg)
+
+
+@api_v2.route('/_error')
+def error():
+    """Implement the endpoint used by httpd, which redirects its errors to it."""
+    try:
+        status = int(os.getenv("REDIRECT_STATUS"))
+    except Exception:
+        # if there's an exception, it means that a client accessed this directly;
+        #  in this case, we want to make it look like the endpoint is not here
+        return api_404_handler()
+    msg = 'Unknown error'
+    if status == 401:
+        msg = 'Authentication failed'
+        raise AuthError(status, msg)
+    elif status == 405:
+        msg = 'Method not allowed for this endpoint'
+    raise HTTPError(status, msg)
+
+
+# flask-restful doesn't actually store a list of endpoints, so we register them as they
+#  pass through add_resource_no_matter_slashes
+
+
+def add_resource_no_matter_slashes(resource, route, endpoint=None, defaults=None):
```

Comment:
Code Reuse

Reply:
@prashbnair if you are referring `api_v1.py` whole idea was to reduce dependency on v1. We do have a far future plan to get rid of v1.